bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/handler/TuyaDeviceHandler.java

```diff
@@ -0,0 +1,658 @@
+
+        TuyaDevice tuyaDevice = this.tuyaDevice;
+        if (tuyaDevice != null) {
+            tuyaDevice.dispose();
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+
+        this.tuyaDevice = new TuyaDevice(gson, this, eventLoopGroup, configuration.deviceId,
+                configuration.localKey.getBytes(StandardCharsets.UTF_8), deviceInfo.ip, deviceInfo.protocolVersion);
+    }
+
+    private void addChannels(Map<String, SchemaDp> schema) {
+        ThingBuilder thingBuilder = editThing();
+        ThingUID thingUID = thing.getUID();
+        ThingHandlerCallback callback = getCallback();
+
+        if (callback == null) {
+            logger.warn("Thing callback not found. Cannot auto-detect thing '{}' channels.", thingUID);
+            return;
+        }
+
+        Map<String, Channel> channels = new HashMap<>(schema.entrySet().stream().map(e -> {
+            String channelId = e.getKey();
+            SchemaDp schemaDp = e.getValue();
+
+            ChannelUID channelUID = new ChannelUID(thingUID, channelId);
+            Map<String, @Nullable Object> configuration = new HashMap<>();
+            configuration.put("dp", schemaDp.id);
+
+            ChannelTypeUID channeltypeUID;
+            if (COLOUR_CHANNEL_CODES.contains(channelId)) {
+                channeltypeUID = CHANNEL_TYPE_UID_COLOR;
+            } else if (DIMMER_CHANNEL_CODES.contains(channelId)) {
+                channeltypeUID = CHANNEL_TYPE_UID_DIMMER;
+                configuration.put("min", schemaDp.min);
+                configuration.put("max", schemaDp.max);
+            } else if ("bool".equals(schemaDp.type)) {
+                channeltypeUID = CHANNEL_TYPE_UID_SWITCH;
+            } else if ("enum".equals(schemaDp.type)) {
+                channeltypeUID = CHANNEL_TYPE_UID_STRING;
+                List<String> range = Objects.requireNonNullElse(schemaDp.range, List.of());
+                configuration.put("range", String.join(",", range));
+            } else if ("string".equals(schemaDp.type)) {
+                channeltypeUID = CHANNEL_TYPE_UID_STRING;
+            } else if ("value".equals(schemaDp.type)) {
+                channeltypeUID = CHANNEL_TYPE_UID_NUMBER;
+                configuration.put("min", schemaDp.min);
+                configuration.put("max", schemaDp.max);
+            } else {
+                // e.g. type "raw", add empty channel
+                return Map.entry("", ChannelBuilder.create(channelUID).build());
+            }
+
+            return Map.entry(channelId, callback.createChannelBuilder(channelUID, channeltypeUID).withLabel(channelId)
+                    .withConfiguration(new Configuration(configuration)).build());
+        }).filter(c -> !c.getKey().isEmpty()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+
+        List<String> channelSuffixes = List.of("", "_1", "_2");
+        List<String> switchChannels = List.of("switch_led", "led_switch");
```

Comment:
It appears that these channels will be dynamically created with `_` as separators. According to our [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention), `-` should be used as the separator.

That said, I understand this binding has a legacy from its previous life at SmartHome/J and already has an existing user base. In light of this, we could consider making an exception here.

However, if it's feasible for existing users to relink their channels, it might be worth evaluating whether the naming convention for channels could still be adhered to. Standardizing now could help maintain consistency across bindings in the long run.